contrib/kube-prometheus/hack/scripts/generate-alertmanager-config-secret.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+cat <<-EOF
+apiVersion: v1
+kind: Secret
+metadata:
+  name: alertmanager-main
+data:
+  alertmanager.yaml: $(cat assets/alertmanager/alertmanager.yaml | base64 --wrap=0)
```

Comment:
There is indeed an equivalent kubectl command, however, this is the [officially documented way](https://kubernetes.io/docs/user-guide/secrets/#creating-a-secret-manually) to do it manually and a `kubectl --dry-run` actually needs a running cluster, which would stop us from running this generate script in our CI pipeline as it's silly to setup a cluster even though we're not actually using it as we're just running `--dry-run` commands. Plus using `kubectl create secret generic --from-file` does not allow setting labels, which we are currently not doing for this secret but for the other `ConfigMap`s.